src/test/ui/const-generics/min_const_generics/invalid-patterns.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(min_const_generics)]
+
+fn get_flag<const FlagSet: bool, const ShortName: char>() -> Option<char> {
+  if FlagSet {
+    Some(ShortName)
+  } else {
+    None
+  }
+}
+
+fn main() {
+  assert!(get_flag::<true, 'c'>().is_some());
+  assert!(get_flag::<false, 'x'>().is_none());
+  get_flag::<false, 0xFF>();
```

Comment:
ah, what I meant here was something like `std::mem::transmute<u8, bool>(3)`, potentially implemented using unions.

So rn you are testing the error messages when using the wrong type, but what we are interested in is the error message when using the right type but an invalid value of said type.

Reply:
oh I see, this path seems very unlikely to get hit then, but worth testing out

Reply:
:laughing: all paths likely to get hit are hopefully already covered by our test suite